api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -402,6 +402,10 @@
      */
     Attribute<Boolean> BUFFER_SPOOLING = new Attribute<Boolean>(true);
     /**
+     * Whether to apply back pressure and stall the upstream operator when the downstream operator is slow.
+     */
+    Attribute<Boolean> BUFFER_BACK_PRESSURE = new Attribute<Boolean>(true);
```

Comment:
I have been thinking about this as well. I went with this becaue of two reasons.

1. There may be scenarios where the input source is bursty and also doesn't have long memory, so large amount of data arrives for a period of time but only remains in the input source for a short period of time after that, so it has to be read out of the input source quickly. In that case it would be better to disable the back pressure and have the data be in our dag and spool it in the input operator for fault tolerance. Because of the bursty nature the output operator will catch up with the input.

2. There may be some unintended issue caused by this fix that we don't see in all our collective wisdom today. At least it will give an opportunity for the user to go back to old behavior.